libdnf/advisory/advisory_query.cpp

```diff
@@ -0,0 +1,437 @@
+                    if (((libsolv_cmp > 0) && ((cmp_type & sack::QueryCmp::GT) == sack::QueryCmp::GT)) ||
+                        ((libsolv_cmp < 0) && ((cmp_type & sack::QueryCmp::LT) == sack::QueryCmp::LT)) ||
+                        ((libsolv_cmp == 0) && ((cmp_type & sack::QueryCmp::EQ) == sack::QueryCmp::EQ))) {
+                        filter_result.add_unsafe((*low).get_advisory_id().id);
+                    }
+                    ++low;
+                }
+            }
+
+        } break;
+        default:
+            throw NotSupportedCmpType("Unsupported CmpType");
+    }
+
+    // Apply filter results to query
+    if (cmp_not) {
+        p_impl->data_map -= filter_result;
+    } else {
+        p_impl->data_map &= filter_result;
+    }
+
+    return *this;
+}
+
+std::vector<AdvisoryPackage> AdvisoryQuery::get_advisory_packages(
+    const libdnf::rpm::PackageSet & package_set, sack::QueryCmp cmp_type) {
+    std::vector<AdvisoryPackage> pkgs = get_sorted_advisory_packages();
+    std::vector<AdvisoryPackage> after_filter;
+
+    auto solv_sack = package_set.get_sack();
+    Pool * pool = solv_sack->p_impl->get_pool();
+
+    switch (cmp_type) {
+        case libdnf::sack::QueryCmp::EQ:
+            //TODO(amatej): faster EQ specific version (we can compare whole NEVRA)
+        case libdnf::sack::QueryCmp::GT:
+        case libdnf::sack::QueryCmp::LT:
+        case libdnf::sack::QueryCmp::GTE:
+        case libdnf::sack::QueryCmp::LTE: {
+            for (libdnf::rpm::PackageSet::iterator package = package_set.begin(); package != package_set.end();
+                 package++) {
+                Solvable * solvable = libdnf::rpm::solv::get_solvable(pool, (*package).get_id().id);
+                auto low = std::lower_bound(pkgs.begin(), pkgs.end(), *package, AdvisoryPackage::comparePackageNA);
+                while (low != pkgs.end() && low->p_impl.get()->get_name_id() == solvable->name &&
+                       low->p_impl.get()->get_arch_id() == solvable->arch) {
+                    int libsolv_cmp = pool_evrcmp(pool, low->p_impl.get()->get_evr_id(), solvable->evr, EVRCMP_COMPARE);
+                    if (((libsolv_cmp > 0) && ((cmp_type & sack::QueryCmp::GT) == sack::QueryCmp::GT)) ||
+                        ((libsolv_cmp < 0) && ((cmp_type & sack::QueryCmp::LT) == sack::QueryCmp::LT)) ||
+                        ((libsolv_cmp == 0) && ((cmp_type & sack::QueryCmp::EQ) == sack::QueryCmp::EQ))) {
+                        after_filter.push_back(*low);
+                    }
+                    ++low;
+                }
+            }
+        } break;
+        default:
+            throw NotSupportedCmpType("Unsupported CmpType");
+    }
+
+    //after_filter contains just advisoryPackages which comply to condition with package_set
+    return after_filter;
+}
+
+std::vector<Advisory> AdvisoryQuery::get_advisories() const {
```

Comment:
May be the method will be unneeded when iterator will be implemented. Then it can be removed or rename to something else. What about to add TODO?

Reply:
There is a TODO at the declaration, but I will add it here as well.